test/NuGet.Core.Tests/NuGet.DependencyResolver.Core.Tests/RemoteWalkContextTests.cs

```diff
@@ -44,19 +44,27 @@ public void FilterDependencyProvidersForLibrary_WhenPackageSourceMappingIsNotEna
             Assert.Equal(context.RemoteLibraryProviders, providers);
         }
 
-        [Fact]
-        public void FilterDependencyProvidersForLibrary_WhenPackageSourceMappingIsEnabledReturnsOnlyApplicableProviders_Success()
+        [Theory]
+        [InlineData("Source1", "Source1")]
+        [InlineData("SOURCE1", "SOURCE1")]
+        [InlineData("Source1", "source1")]
+        [InlineData("source1", "Source1")]
+        [InlineData("SOURCE1", "Source1")]
+        [InlineData("Source1", "SOURCE1")]
+        public void FilterDependencyProvidersForLibrary_WithDifferentCasing_WhenPackageSourceMappingIsEnabledReturnsOnlyApplicableProviders_Success(string packageSourceMappingName, string providerName)
```

Comment:
The inputs are really a mappingSource and sourceName, right?
Also, can the name be more succinct?

```suggestion
        public void FilterDependencyProvidersForLibrary_LookupSourcesWithVariousCasing_ReturnsOnlyApplicableProviders(string packageSourceMappingName, string source)
```

Reply:
In this case the type being returned is a IRemoteDependencyProvider (which despite the name, can be a local source), so I think provider name is more appropriate.

I did incorporate some of the feedback in the newest commit.